Workbooks/Azure API Management/APIMAnalytics/APIMAnalytics.workbook

```diff
@@ -0,0 +1,567 @@
+              "size": 0,
+              "queryType": 0,
+              "resourceType": "microsoft.apimanagement/service",
+              "mapSettings": {
+                "locInfo": "CountryRegion",
+                "locInfoColumn": "Region",
+                "sizeSettings": "Region",
+                "sizeAggregation": "Sum",
+                "legendMetric": "SuccessFullRequests",
+                "legendAggregation": "Sum",
+                "itemColorSettings": {
+                  "nodeColorField": "SuccessFullRequests",
+                  "colorAggregation": "Sum",
+                  "type": "heatmap",
+                  "heatmapPalette": "greenRed"
+                }
+              }
+            },
+            "name": "query - 0"
+          }
+        ]
+      },
+      "conditionalVisibility": {
+        "parameterName": "tab",
+        "comparison": "isEqualTo",
+        "value": "geography"
+      },
+      "name": "Geography"
+    },
+    {
+      "type": 12,
+      "content": {
+        "version": "NotebookGroup/1.0",
+        "groupType": "editable",
+        "items": [
+          {
+            "type": 3,
+            "content": {
+              "version": "KqlItem/1.0",
+              "query": "ApiManagementGatewayLogs\r\n| summarize \r\n            SuccessFullRequests = countif(ResponseCode < 400), \r\n            FailedRequests = countif(ResponseCode == 400 or (ResponseCode >= 500 and ResponseCode < 600)),\r\n            UnauthorizedRequests = countif(ResponseCode == 401 or ResponseCode == 403 or ResponseCode == 429),\r\n            OtherRequests=countif((ResponseCode > 401 and ResponseCode < 500 and ResponseCode != 401 and ResponseCode != 403 and ResponseCode != 429) or ResponseCode > 599 or  isempty(ResponseCode)),\r\n            TotalRequests=count(),\r\n            AverageResponseTime=strcat(tostring(round(avgif(TotalTime, TotalTime > 0), 2)), \" ms\"),\r\n            AverageServiceResponseTime=strcat(tostring(round(avgif(BackendTime, BackendTime > 0), 2)), \" ms\"),\r\n            DataTransfer=strcat(tostring(sum(ResponseSize) / 1000), \" KB\") by ApiId, APIid = strcat(\"/apis/ \", ApiId)",
```

Comment:
`AverageResponseTime=strcat(tostring(round(avgif(TotalTime, TotalTime > 0), 2)), \" ms\")`

instead of formatting your numbers to strings, you can leave them as numbers, and then in the grid settings use the number format settings to set the number of decimals and the units.  doing it here in the query and returning it as text means that people in locales that use `,` instead of `.` will get formatted incorrectly to them.  Using the units in the grid settings instead also lets this scale up, large values will get rendered as seconds, minutes, days, etc, and sorting/etc will work as numbers

Reply:
Thanks for the review I changed as you suggested used column settings for formatting data